crates/sui/src/upgrade_compatibility.rs

```diff
@@ -1326,15 +1381,27 @@ fn struct_field_mismatch_diag(
 
     let def_loc = struct_sourcemap.definition_location;
 
-    if old_struct.fields.len() != new_struct.fields.len() {
+    let old_fields: Vec<&Field> = old_struct
+        .fields
+        .iter()
+        .filter(|f| f.name != Identifier::new("dummy_field").unwrap() && f.type_ != Type::Bool)
```

Comment:
confused a bit about the `dummy_field` identifier. Is that a common identifier that exists everywhere?

Reply:
it comes from the move compiler, if you do not have a field then you get a dummy field since the compiler needs *something* in the type for it to exist. The compiler makes a `dummy_field: bool` in this case.